letter-sender/src/test/java/com/christmas/letter/sender/controller/LetterControllerTest.java

```diff
@@ -0,0 +1,81 @@
+package com.christmas.letter.sender.controller;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.christmas.letter.sender.model.Letter;
+import com.christmas.letter.sender.model.Location;
+import com.christmas.letter.sender.service.LetterService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+@WebMvcTest(LetterController.class)
+class LetterControllerTest {
+
+  @SuppressWarnings("SpringJavaInjectionPointsAutowiringInspection")
+  @Autowired
+  private MockMvc mockMvc;
+
+  @MockBean
+  private LetterService letterService;
+
+  private static final String PATH = "/api/letters";
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  @Test
+   void When_CreateLetter_Expect_LetterSend() throws Exception {
+
+    mockMvc.perform(post(PATH)
+        .contentType(MediaType.APPLICATION_JSON)
+        .content(objectMapper.writeValueAsString(createLetter())))
+        .andExpect(status().isCreated());
+
+  }
+
+ @Test
+  void When_CreateLetter_Expect_BadRequest() throws Exception {
+
+    String invalidTestMessage = objectMapper.writeValueAsString(createInvalidLetter());
+
+    mockMvc.perform(post(PATH)
```

Comment:
Let's also verify that the message returned by the api is helping the user correct the payload, so let's verify that it actually says "email is required" etc etc

Reply:
check .andExpect(jsonPath("$.message").value("The validation has failed!"));